etc/kayobe/ansible/roles/opensearch/tasks/set_log_retention_policy.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Check required variables are set
+  fail:
+    msg: "Please set the variable {{ item.key }} in opensearch.yml. For example: {{ item.key }}: {{ item.value }} "
+  when: "{{ item.key }} is not defined"
+  with_dict:
+    - "opensearch_internal_endpoint": '"http://{{ kolla_internal_fqdn }}:9200"'
+    - "opensearch_log_index_pattern": '"flog-*"'
+
+- name: Wait for OpenSearch to become ready
+  become: true
+  uri:
+    url: "{{ opensearch_internal_endpoint }}/_cluster/stats"
```

Comment:
Running on controllers should also (usually) avoid the issue of CA certs for internal TLS, since the vault playbooks add them to the trust store on controllers.